rs/nns/integration_tests/src/canister_upgrade.rs

```diff
@@ -24,13 +25,13 @@ fn test_upgrade_canister(canister_id: CanisterId, canister_wasm: Wasm, use_propo
 
     let modified_wasm = modify_wasm_bytes(&canister_wasm.bytes(), 42);
 
-    let _proposal_id = nns_propose_upgrade_nns_canister(
+    let proposal_id = nns_propose_upgrade_nns_canister(
         &state_machine,
         n1.principal_id,
         n1.neuron_id,
         canister_id,
         modified_wasm.clone(),
-        Encode!(&()).unwrap(),
+        vec![],
```

Comment:
So there's a semantic difference I think between encoding a unit and providing an empty vector (at least there was).

Does the code account for the way this is ultimately passed to the canister getting upgraded?

Reply:
Because of [this](https://github.com/dfinity/ic/blob/2a2b74684b55c40d514c2e80165432e906dddbf5/testnet/tools/nns-tools/submit-mainnet-nns-upgrade-proposal.sh#L86) and [this](https://github.com/dfinity/ic/blob/2a2b74684b55c40d514c2e80165432e906dddbf5/rs/registry/admin/src/main.rs#L1068), we currently are using `vec![]`, not `Encode!(&()).unwrap()` for NNS canisters without upgrade args. I can avoid changing it here, but I felt it's more realistic to have `vec![]` here.